render_delegate/material.cpp

```diff
@@ -201,10 +201,17 @@ RemapNodeFunc floatPrimvarRemap = [](MaterialEditContext* ctx) {
 // have to use a utility node instead of a user_data_rgb node.
 RemapNodeFunc float2PrimvarRemap = [](MaterialEditContext* ctx) {
     const auto varnameValue = ctx->GetParam(str::t_varname);
-    if (varnameValue.IsEmpty() || !varnameValue.IsHolding<TfToken>()) {
-        return;
+    TfToken varname;
+    if (varnameValue.IsHolding<TfToken>()) {
+        varname = varnameValue.UncheckedGet<TfToken>();
+	} else if (varnameValue.IsHolding<std::string>()) {
+        varname = pxr::TfToken(varnameValue.UncheckedGet<std::string>());
     }
-    const auto& varname = varnameValue.UncheckedGet<TfToken>();
+
+	if (varname.IsEmpty()) {
```

Comment:
I think it is a mistake in the original code to return early here. We should always produce a valid node, otherwise, we would try to translate a non-existing node (UsdPrimvarReader_float2). Let's just skip the return and the check for the varname is empty. Arnold will warn the user about the empty varname, like with the other primvar readers if the user is using an incomplete reader node.

Reply:
Then leaves us with two options:
- falling back to reading the UVs from the mesh if any
- creating a user_data_rgb node with an empty attribute name

Going for the second one where Arnold will issue an error message.